### PR TITLE
create NVMe config files before udevd is started (bsc#1184908)

### PR DESCRIPTION
**Pull request by wfeldt**
## Problem

- https://bugzilla.suse.com/show_bug.cgi?id=1184908#c78

The nvme config is created too late for fc nvme discovery to work correctly.

## Solution

Move the nvme config creation into the `udev_setup` script.

### CODE REVIEW

File: data/initrd/scripts/udev_setup
Comment (igaw) [inline data/initrd/scripts/udev_setup:30]: While at it, could we also test if the hostnqn file already exists? If it does, just skip this step? 
This would allow to inject a hostnqn via the command line, e.g. 

`  echo nqn.2014-08.org.nvmexpress:uuid:1a9e23dd-466e-45ca-9f43-a29aaf47cb21 > /etc/nvme/hostnqn
`

**Review by igaw — CHANGES_REQUESTED** [review]

File: data/initrd/scripts/udev_setup
Comment (wfeldt) [reply data/initrd/scripts/udev_setup:30]: Command line is actually parsed later. For this to work properly we'd really need a separate linuxrc option.

Comment (igaw) [reply data/initrd/scripts/udev_setup:30]: Understood. I've modified the line to 

`if [ ! -f /etc/nvme/hostnqn ] && [ -x /usr/sbin/nvme ] ; then`

and added the hostnqn file to initrd via mksusecd. My test setup with qemu doesn't have
have proper interfaces to generate the correct system uuid. Anyway with this I was
able to get it working.

**Comment by wfeldt** [comment]
Ok, added the test for an existing hostnqn file. Note that this also skips the hostid generation.

**Comment by igaw** [comment]
> Ok, added the test for an existing hostnqn file. Note that this also skips the hostid generation.

Should be okay, as if the user adds the hostnqn we can assume the hostid file also exists. I think this a fair assumption.

**Comment by igaw** [comment]
Alright this looks good to me. I really hope we got it finally correct.

**Comment by wfeldt** [comment]
As we're not in a hurry atm, I'd like to wait for feedback from Ricky before merging.

**Pull request merged**
